### PR TITLE
[FIX] Gas is not re-calculated when updating a transaction

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**
There is a problem with gas calculation when we switch from one type of Send to another (Send ETH/ Send ERC20 / Send NFT). The transaction object retains the previous gas value before attempting to estimate the new gas value. This bug results in non-desired behaviours for both cases.

**Screenshots/Recordings**
Before
https://recordit.co/oGbCf2W6iy

After
http://recordit.co/xOnFZ3DyQy

**Issue**

Progresses #5865 

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/util/custom-gas/customGas.test.ts [hunk 1]
```diff
@@ -26,3 +37,15 @@ describe('CustomGas utils :: parseWaitTime', () => {
...
+  it('should return passed gas value', async () => {
+    const estimate = await getGasLimit({ gas: '0xF4240', gasPrice: '12' });
+    expect(estimate.gas.toNumber()).toEqual(21000);
+  });
+
+  it('should fetch new estimated gas value', async () => {
+    const estimate = await getGasLimit({ gas: '0x5208', gasPrice: '12' }, true);
+    expect(estimate.gas.toNumber()).toEqual(21000);
+  });
+});
```
Comment (jpuri) [inline app/util/custom-gas/customGas.test.ts:51]: Hey both scenario have 21000 gas, that looks wrong.

Comment (blackdevelopa) [reply app/util/custom-gas/customGas.test.ts:54]: Updated now. https://github.com/MetaMask/metamask-mobile/pull/5876/commits/cc9f732c2a25f44782665a1158962adbe1b3098a

**Review by jpuri — APPROVED** [review]
Good work 👍

**Review by NiranjanaBinoy — APPROVED** [review]
lgtm

**Comment by seaona** [comment]
PR looks good @blackdevelopa :100:  I've checked switching from ERC20 to ETH and the other way around as well as with ERC721 tokens and they all work fine.

A small thing I've noticed is that the button Send is enabled before the gas has updated, but this can be handled in the future, whenever the loading states are consolidated. cc @bschorchit 

The PR can be merged, just requires to fix some conflicts with a test file.

https://user-images.githubusercontent.com/54408225/235098259-9b608f89-9c86-40cc-9e0d-3bf5458c7109.mp4

**Comment by seaona** [comment]
Re-tested again and looks good. I think we can merge it since last changes were due to conflict on a snapshot test

**Pull request merged**